added credit  usage warning on 80%

itzamanjain opened the pull request:
## Description ✏️

fix [#18](https://github.com/trypear/pearai-server-issues-public/issues/18)

What changed? Feel free to be brief.
Implemented functionality to fetch warning message from the server. (backend)
Display  warning when usage exceeds 80%, helping users stay informed about their credit consumption.

- Bullet points are helpful.
- Screenshots are helpful (if applicable).
![Screenshot 2024-10-29 194228](https://github.com/user-attachments/assets/e6a86e0e-e032-42bf-b681-1677bae6bad7)


## Checklist ✅

- [x] I have added screenshots (if UI changes are present).
- [x] I have done a self-review of my code.
- [ ] I have manually tested my code (if applicable).

nang-dev commented on core/config/default.ts:
```diff
@@ -24,6 +24,11 @@ export const defaultConfig: SerializedContinueConfig = {
       provider: "free-trial",
       model: "claude-3-sonnet-20240229",
     },
```
This is perfect, but some Claude 3.5 changes here D:

cqdev-co commented:
@nang-dev I think I accidentally made these changes off of cqdev/claude3.5 instead of main, so once 3.5 gets pushed then there will be less changes from this. If I screwed up I don't mind deleting this pr and branch then doing this properly (oopsie), just let me know what you wanna do!

cqdev-co commented:
Deleting branch and closing

itzamanjain committed:
added credit  usage warning on 80%

itzamanjain commented:
looking for how to add native vs code notifcation

itzamanjain committed:
added vscode native notification warning for credit used more than 80%

cqdev-co commented:
@itzamanjain how's it going on this

itzamanjain commented:
@cqdev-co  I added a VSCode native notification warning for it, along with our own custom notification, but I wasn't able to test the native notification. I also added logic to the server's /get-usage API, so it now returns a warningMessage if the credit is more than 80%, otherwise it returns WarningMessage : None.

itzamanjain committed:
usertoken

Himanshu-Singh-Chauhan commented on gui/src/components/usagesWarning.tsx:
```diff
@@ -0,0 +1,133 @@
+      if (warning_message) {
+        setMessage(warning_message);
+        setIsVisible(true); 
+        vscode.window.showWarningMessage(warning_message);
```
does vscode.window work this way in react 👀 isnt working for me

itzamanjain replied:
based on prev patterns used in code base i used this so we can native notification warning

Himanshu-Singh-Chauhan commented on gui/src/components/usagesWarning.tsx:
```diff
@@ -0,0 +1,133 @@
+    }
+  };
+
+  useEffect(() => {
+    getMessage();
+  }, []);
```
this will only fetch when component gets mounted, so if users is overusing credits it in a single go in one session, he wouldn't get notified right.

itzamanjain replied:
>Thinking of using setInterval to poll every 5 minutes for credit usage—open to better ideas if anyone has them!
